### DISCUSSION TITLE
Security Bundle : CheckCredentials in AbstractLoginFormAuthenticator

### DISCUSSION THREAD

**Discussion by nicosapp** [symfony/symfony#58018] [Q&A]
Hello folks!
I try to implement a custom "checkCredentials" for the AbstractLoginFormAuthenticator but as mentionned here https://github.com/symfony/symfony/issues/40869, it is no longer in this class.
Do you know how can I do?
Thanks!

GitHubGitHub
[AbstractGuardAuthenticator not implementing the right interface (AuthenticatorInterface) causing AuthenticatorManager::executeAuthenticator() to throw an error ?! · Issue #40869 · symfony/symfony](https://github.com/symfony/symfony/issues/40869)

**Comment by MatTheCat** [comment]
Hello,

`checkCredentials` was a method of the guard authenticator interface which has been deprecated in Symfony 5.3: https://symfony.com/blog/new-in-symfony-5-3-guard-component-deprecation

Look at the current documentation to implement your login form: https://symfony.com/doc/current/security.html#form-login

**Reply by nicosapp** [reply]
I have already checked it and I did not find my answer :( .
If I want to use "checkCredentials" it is because I migrate a symfony 3.3 project to 7.1
In my project (3.3), password are hashed with sha512 + base64 + salt.

I tried also this documentation => https://symfony.com/doc/current/security/passwords.html#configure-a-new-hasher-using-migrate-from
But it does not work.

For info,
- to get base64 sha512 hashed password => $user->getPassword()
- to get salt => $user->getSalt()

Thanks for your reply

**Comment by nicosapp** [comment]
please help :pray:
